src/openvic-simulation/types/IdentifierRegistry.hpp

```diff
@@ -315,98 +315,85 @@ namespace OpenVic {
 	template<std::derived_from<HasIdentifier> _Type>
 	using IdentifierInstanceRegistry = InstanceRegistry<_Type, _get_identifier<_Type>>;
 
-/* Member functions for const access to a UniqueKeyRegistry member variable. */
-#define IDENTIFIER_REGISTRY_ACCESSORS(name) \
-	IDENTIFIER_REGISTRY_ACCESSORS_CUSTOM_PLURAL(name, name##s)
+/* Macros to generate declaration and constant accessor methods for a UniqueKeyRegistry member variable. */
 
-#define IDENTIFIER_REGISTRY_ACCESSORS_CUSTOM_PLURAL(singular, plural) \
-	IDENTIFIER_REGISTRY_ACCESSORS_FULL_CUSTOM(singular, plural, plural, 0)
+#define IDENTIFIER_REGISTRY(registry_type, name) \
+	IDENTIFIER_REGISTRY_CUSTOM_PLURAL(registry_type, name, name##s)
 
-#define IDENTIFIER_REGISTRY_ACCESSORS_CUSTOM_INDEX_OFFSET(name, index_offset) \
-	IDENTIFIER_REGISTRY_ACCESSORS_FULL_CUSTOM(name, name##s, name##s, index_offset)
+#define IDENTIFIER_REGISTRY_CUSTOM_PLURAL(registry_type, singular, plural) \
+	IDENTIFIER_REGISTRY_FULL_CUSTOM(registry_type, singular, plural, plural, plural, 0)
 
-#define IDENTIFIER_REGISTRY_ACCESSORS_FULL_CUSTOM(singular, plural, registry, index_offset) \
+#define IDENTIFIER_REGISTRY_CUSTOM_INDEX_OFFSET(registry_type, name, index_offset) \
+	IDENTIFIER_REGISTRY_FULL_CUSTOM(registry_type, name, name##s, name##s, name##s, index_offset)
+
+#define IDENTIFIER_REGISTRY_FULL_CUSTOM(registry_type, singular, plural, registry, debug_name, index_offset) \
+	registry_type plural { #debug_name };\
+public: \
 	void lock_##plural() { \
 		registry.lock(); \
 	} \
-	bool plural##_are_locked() const { \
+	bool plural##_are_locked() \
+	const \
+	{ \
```

Comment:
```suggestion
	bool plural##_are_locked() const { \
```
Also can you move these onto the same line